fc/mef.py

```diff
@@ -736,230 +506,223 @@ def get_transform_fxn(data_beads, peaks_mef, mef_channels,
     if plot_filename is None:
         plot_filename = str(data_beads)
 
-    # 1. Cluster
-    # ===========
-    if cluster_method == 'dbscan':
-        labels = clustering_dbscan(data_beads[:,cluster_channels], 
-            **cluster_params)
-    elif cluster_method == 'distance':
-        labels = clustering_distance(data_beads[:,cluster_channels], 
-            **cluster_params)
-    elif cluster_method == 'gmm':
-        labels = clustering_gmm(data_beads[:,cluster_channels], 
-            **cluster_params)
+    # mef_channels and mef_values should be iterables.
+    if hasattr(mef_channels, '__iter__'):
+        mef_channel_all = list(mef_channels)
+        mef_values_all = np.array(mef_values).copy()
     else:
-        raise ValueError("Clustering method {} not recognized."
-            .format(cluster_method))
+        mef_channel_all = [mef_channels]
+        mef_values_all = np.array([mef_values])
```

Comment:
I don't think there is a situation in which they don't have the same number of peaks. Also, given that clustering can be made with several channels, we definitely need the same number of populations in all of these channels.